### PR TITLE
Update mod.rs

### PR DESCRIPTION
**Pull request by austineinstein**
The Shader program.

### CODE REVIEW

File: src/Shaders/mod.rs [hunk 1]
```diff
@@ -160,5 +160,40 @@ impl Drop for FragmentShader {
...
+    }
+}
+
+impl linker for ShaderProgram {
+    fn compile(&mut self) {
+       unsafe{gl::AttachShader(self.m_ID, gl::VERTEX_SHADER)}; 
+       unsafe{gl::AttachShader(self.m_ID, gl::FRAGMENT_SHADER)};
+       unsafe{gl::LinkProgram(self.m_ID)};
+       let compileStatus : ShaderCompile = unsafe {
+           gl::GetProgramiv(self.m_ID, gl::COMPILE_STATUS, *mut success as *mut c_uint);
```
Comment (austineinstein) [inline src/Shaders/mod.rs]: c_int*

File: src/Shaders/mod.rs [hunk 2]
```diff
@@ -49,8 +49,8 @@ impl shader for VertexShader {
 
     fn compile(&self) -> ShaderCompile {
 
-        let SourceLenght  = self.m_Source.len() as i32;
```
Comment (glowing-chemist) [inline src/Shaders/mod.rs]: before commiting you should make sure your not accidentally changing lines you don't mean to.

File: src/Shaders/mod.rs [hunk 3]
```diff
@@ -160,5 +160,40 @@ impl Drop for FragmentShader {
 
 
 pub struct ShaderProgram {
+    m_ID : u32
```
Comment (glowing-chemist) [inline src/Shaders/mod.rs]: The shader program should contain the vertex and fragment shaders it was made from idealy.

File: src/Shaders/mod.rs [hunk 4]
```diff
@@ -160,5 +160,40 @@ impl Drop for FragmentShader {
...
+}
+
+impl ShaderProgram {
+    fn new(path : String) -> ShaderProgram {
```
Comment (glowing-chemist) [inline src/Shaders/mod.rs]: the "constructor" shoul take a vertex and fragment shader.

File: src/Shaders/mod.rs [hunk 5]
```diff
@@ -160,5 +160,40 @@ impl Drop for FragmentShader {
...
+        let ID = unsafe{gl::CreateProgram()};
+        ShaderProgram{m_ID : ID}
+    }
+    fn Get_ID(&self) -> u32 {
+        self.m_ID
+    }
+}
+
+impl linker for ShaderProgram {
```
Comment (glowing-chemist) [inline src/Shaders/mod.rs]: Linker isn't a trait defined ithis module.

File: src/Shaders/mod.rs [hunk 6]
```diff
@@ -160,5 +160,40 @@ impl Drop for FragmentShader {
...
+    fn compile(&mut self) {
```
Comment (glowing-chemist) [inline src/Shaders/mod.rs]: This function should really be called link as thats what it is doing. Also in the calls to AttachShader the ID of the respective shaders should be used. One last thing the functiuon doesn't need to take self by mutanle reference.

File: src/Shaders/mod.rs (hunk 2 above)
Comment (austineinstein) [reply src/Shaders/mod.rs]: I did mean to change your spelling of length as 'lenght'

**Pull request closed without merging**